src/main/java/org/jenkinsci/plugins/p4/scm/StreamsScmSource.java

```diff
@@ -0,0 +1,57 @@
+package org.jenkinsci.plugins.p4.scm;
+
+import com.perforce.p4java.core.IStreamSummary;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.p4.browsers.P4Browser;
+import org.jenkinsci.plugins.p4.client.ConnectionHelper;
+import org.jenkinsci.plugins.p4.workspace.StreamWorkspaceImpl;
+import org.jenkinsci.plugins.p4.workspace.Workspace;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+public class StreamsScmSource extends AbstractP4ScmSource {
```

Comment:
Thus separation may be excessive. may be a good idea to have just one source rather than multiple, otherwise you risk complications when using from an SCMNavigator